packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -77,17 +78,141 @@ enum SystemUiOverlay {
   bottom,
 }
 
+/// Describes the contrast needs of a color.
+enum Brightness {
+  /// The color is dark and will require a light text color to achieve readable
+  /// contrast.
+  ///
+  /// For example, the color might be dark grey, requiring white text.
+  dark,
+
+  /// The color is light and will require a dark text color to achieve readable
+  /// contrast.
+  ///
+  /// For example, the color might be bright white, requiring black text.
+  light,
+}
+
 /// Specifies a preference for the style of the system overlays.
 ///
 /// Used by [SystemChrome.setSystemUIOverlayStyle].
-enum SystemUiOverlayStyle {
+class SystemUiOverlayStyle {
   /// System overlays should be drawn with a light color. Intended for
   /// applications with a dark background.
-  light,
+  static const SystemUiOverlayStyle light = const SystemUiOverlayStyle(
+    systemNavigationBarColor: const Color(0xFFFFFFFF),
+    systemNavigationBarDividerColor: null,
+    statusBarColor: null,
+    systemNavigationBarIconBrightness: Brightness.dark,
+    statusBarIconBrightness: Brightness.dark,
+    statusBarBrightness:  Brightness.dark,
+  );
 
   /// System overlays should be drawn with a dark color. Intended for
   /// applications with a light background.
-  dark,
+  static const SystemUiOverlayStyle dark = const SystemUiOverlayStyle(
+    systemNavigationBarColor: const Color(0xFF000000),
+    systemNavigationBarDividerColor: null,
+    statusBarColor: null,
+    systemNavigationBarIconBrightness: Brightness.light,
+    statusBarIconBrightness: Brightness.light,
+    statusBarBrightness: Brightness.light,
+  );
+
+  /// Creates a new [SystemUiOverlayStyle].
+  const SystemUiOverlayStyle({
+    this.systemNavigationBarColor,
+    this.systemNavigationBarDividerColor,
+    this.statusBarColor,
+    this.statusBarBrightness,
+    this.statusBarIconBrightness,
+    this.systemNavigationBarIconBrightness,
+  });
+
+  /// The color of the system bottom navigation bar.
+  /// 
+  /// Only honored in Android versions O and greater.
+  final Color systemNavigationBarColor;
+
+  /// The color of the divider between the system's bottom navigation bar and the app's content.
+  /// 
+  /// Only honored in Android versions P and greater.
+  final Color systemNavigationBarDividerColor;
+
+  /// The color of top status bar.
+  /// 
+  /// Only honored in Android version M and greater.
+  final Color statusBarColor;
+
+  /// The brightness of top status bar.
+  /// 
+  /// Only honored in iOS .
+  final Brightness statusBarBrightness;
+
+  /// The brightness of the top status bar icons.
+  /// 
+  /// Only honored in Android version M and greater.
+  final Brightness statusBarIconBrightness;
+
+  /// The brightness of the system navigation bar icons.
```

Comment:
It would be a little tidier if this definition was next to the other system nav bar defs

Reply:
Done